# QUESTION 1

A F&B manager wants to determine whether there is any significant difference in the diameter of the cutlet between two units. A randomly selected sample of cutlets was collected from both units and measured? Analyze the data and draw inferences at 5% significance level. Please state the assumptions and tests that you carried out to check validity of the assumptions.


     Minitab File : Cutlets.mtw


In [4]:
# Importing Dataset
import pandas as pd
cutlets=pd.read_csv("Cutlets.csv")
cutlets

Unit A  Unit B
0   6.8090  6.7703
1   6.4376  7.5093
2   6.9157  6.7300
3   7.3012  6.7878
4   7.4488  7.1522
5   7.3871  6.8110
6   6.8755  7.2212
7   7.0621  6.6606
8   6.6840  7.2402
9   6.8236  7.0503
10  7.3930  6.8810
11  7.5169  7.4059
12  6.9246  6.7652
13  6.9256  6.0380
14  6.5797  7.1581
15  6.8394  7.0240
16  6.5970  6.6672
17  7.2705  7.4314
18  7.2828  7.3070
19  7.3495  6.7478
20  6.9438  6.8889
21  7.1560  7.4220
22  6.5341  6.5217
23  7.2854  7.1688
24  6.9952  6.7594
25  6.8568  6.9399
26  7.2163  7.0133
27  6.6801  6.9182
28  6.9431  6.3346
29  7.0852  7.5459
30  6.7794  7.0992
31  7.2783  7.1180
32  7.1561  6.6965
33  7.3943  6.5780
34  6.9405  7.3875

'''null hypothesis  =  there is no any significant difference in the diameter of the cutlet between two units

alternative hypothesis =  there is any significant difference in the diameter of the cutlet between two units'''

In [6]:
unitA=pd.Series(cutlets.iloc[:,0])
unitA

0     6.8090
1     6.4376
2     6.9157
3     7.3012
4     7.4488
5     7.3871
6     6.8755
7     7.0621
8     6.6840
9     6.8236
10    7.3930
11    7.5169
12    6.9246
13    6.9256
14    6.5797
15    6.8394
16    6.5970
17    7.2705
18    7.2828
19    7.3495
20    6.9438
21    7.1560
22    6.5341
23    7.2854
24    6.9952
25    6.8568
26    7.2163
27    6.6801
28    6.9431
29    7.0852
30    6.7794
31    7.2783
32    7.1561
33    7.3943
34    6.9405
Name: Unit A, dtype: float64

In [8]:
unitB=pd.Series(cutlets.iloc[:,1])
unitB

0     6.7703
1     7.5093
2     6.7300
3     6.7878
4     7.1522
5     6.8110
6     7.2212
7     6.6606
8     7.2402
9     7.0503
10    6.8810
11    7.4059
12    6.7652
13    6.0380
14    7.1581
15    7.0240
16    6.6672
17    7.4314
18    7.3070
19    6.7478
20    6.8889
21    7.4220
22    6.5217
23    7.1688
24    6.7594
25    6.9399
26    7.0133
27    6.9182
28    6.3346
29    7.5459
30    7.0992
31    7.1180
32    6.6965
33    6.5780
34    7.3875
Name: Unit B, dtype: float64

In [10]:
import scipy.stats as stats
p_value=stats.ttest_ind(unitA,unitB)
p_value

Ttest_indResult(statistic=0.7228688704678063, pvalue=0.4722394724599501)

In [11]:
p_value[1]

0.4722394724599501

As our p_value value is less than alpha So we reject the null hypothesis.

So, there is any significant difference in the diameter of the cutlet between two units. 

# QUESTION 2

A hospital wants to determine whether there is any difference in the average Turn Around Time (TAT) of reports of the laboratories on their preferred list. They collected a random sample and recorded TAT for reports of 4 laboratories. TAT is defined as sample collected to report dispatch.
   
  Analyze the data and determine whether there is any difference in average TAT among the different laboratories at 5% significance level.
 
 
    Minitab File: LabTAT.mtw


In [15]:
LabTAT=pd.read_csv("LabTAT.csv")
LabTAT
pd.set_option("display.max_rows",None)
LabTAT

Laboratory 1  Laboratory 2  Laboratory 3  Laboratory 4
0          185.35        165.53        176.70        166.13
1          170.49        185.91        198.45        160.79
2          192.77        194.92        201.23        185.18
3          177.33        183.00        199.61        176.42
4          193.41        169.57        204.63        152.60
5          179.45        197.00        181.51        161.12
6          191.37        166.36        214.21        154.02
7          166.81        169.60        183.43        163.25
8          158.81        175.36        191.60        152.79
9          165.88        198.68        208.43        161.98
10         174.75        189.12        231.00        171.22
11         193.37        140.55        198.37        183.67
12         184.75        160.44        226.62        142.95
13         178.54        167.03        214.44        152.37
14         180.19        182.67        159.69        163.81
15         172.17        155.73        214.14        156.06
16         172.00        183.07        212.29        176.44
17         184.92        177.70        209.25        173.68
18         187.38        191.62        198.54        161.49
19         194.17        186.85        229.02        175.05
20         192.03        161.09        214.71        150.22
21         193.77        173.01        172.27        148.17
22         178.24        199.43        205.31        166.21
23         171.41        185.67        180.35        145.40
24         178.30        182.69        214.58        166.42
25         176.13        180.45        231.76        132.21
26         180.67        182.37        215.78        143.26
27         177.38        189.99        201.78        154.06
28         190.36        167.55        191.64        179.08
29         199.99        190.09        217.56        144.82
30         181.36        168.01        193.04        184.64
31         187.37        176.50        217.66        188.49
32         172.93        173.54        198.68        198.69
33         187.18        153.75        206.49        142.43
34         179.03        212.80        198.81        169.68
35         181.69        179.32        200.58        174.79
36         185.28        173.56        199.91        133.31
37         157.14        217.86        190.34        176.42
38         184.42        187.57        181.87        176.67
39         183.45        163.69        214.39        158.20
40         175.50        186.49        189.61        157.02
41         169.55        171.31        203.58        154.55
42         166.86        202.29        196.64        157.58
43         166.33        197.68        225.42        161.96
44         173.88        156.89        175.13        168.14
45         189.84        168.03        228.21        180.14
46         184.88        173.09        229.33        170.28
47         188.63        187.52        170.58        166.68
48         216.39        193.28        177.76        161.42
49         180.57        178.04        207.34        153.97
50         177.54        197.45        178.07        167.50
51         182.34        172.35        188.32        165.43
52         187.35        167.63        193.26        150.83
53         185.87        194.24        201.17        149.96
54         188.01        179.43        192.75        161.94
55         182.62        183.62        192.81        146.46
56         179.08        157.92        183.10        174.83
57         189.96        173.55        181.92        181.66
58         169.15        180.86        209.43        164.82
59         169.83        161.02        180.11        181.35
60         181.45        178.42        172.92        176.21
61         196.13        165.12        205.01        152.40
62         172.02        183.75        206.46        159.63
63         181.83        186.77        192.48        156.56
64         170.41        177.59        237.49        168.11
65         161.99        202.78        173.08

null hypothesis =  there is no any difference in average TAT among the different laboratories

alternative hypothesis =  there is any difference in average TAT among the different laboratories

In [17]:
import scipy.stats as stats
from scipy.stats import f_oneway
f_oneway(LabTAT["Laboratory 1"], LabTAT["Laboratory 2"], LabTAT["Laboratory 3"], LabTAT["Laboratory 4"])

F_onewayResult(statistic=118.70421654401437, pvalue=2.1156708949992414e-57)

P_value is greater than alpha we accept null hypothesis

So we can say there is no any difference in average TAT among the different laboratories

# QUESTION 3

Sales of products in four different regions is tabulated for males and females. Find if male-female buyer rations are similar across regions.

In [20]:
# importing dataset
buyerratio=pd.read_csv("BuyerRatio.csv")
buyerratio

Observed Values  East  West  North  South
0           Males    50   142    131     70
1         Females   435  1523   1356    750

Null Hypothesis: All proportions are equal

Alternative Hypothesis: Not all Proportions are equal

In [34]:
buyerratio1=buyerratio.iloc[0:,1:]
buyerratio1

East  West  North  South
0    50   142    131     70
1   435  1523   1356    750

In [35]:
# Changing the data frame format to array format
data = buyerratio1.to_numpy()
data

array([[  50,  142,  131,   70],
       [ 435, 1523, 1356,  750]], dtype=int64)

In [36]:
# To find the Chi square value and p_value
import scipy.stats as stats
from scipy.stats import chi2_contingency
chi2_contingency(data)

(1.595945538661058,
 0.6603094907091882,
 3,
 array([[  42.76531299,  146.81287862,  131.11756787,   72.30424052],
        [ 442.23468701, 1518.18712138, 1355.88243213,  747.69575948]]))

Chi-square value = 1.5959

p_value = 0.6603

degree of freedom = 3

alpha value = 0.05

As the p_value is greater than alpha value we accept the null hypothesis.
So we can say all propertions are equal.

# QUESTION 4

TeleCall uses 4 centers around the globe to process customer order forms. They audit a certain %  of the customer order forms. Any error in order form renders it defective and has to be reworked before processing.  The manager wants to check whether the defective %  varies by centre. Please analyze the data at 5% significance level and help the manager draw appropriate inferences

Minitab File: CustomerOrderForm.mtw
 
     

In [39]:
# importing dataset
customer=pd.read_csv("Costomer+OrderForm.csv")
customer

Phillippines   Indonesia       Malta       India
0     Error Free  Error Free   Defective  Error Free
1     Error Free  Error Free  Error Free   Defective
2     Error Free   Defective   Defective  Error Free
3     Error Free  Error Free  Error Free  Error Free
4     Error Free  Error Free   Defective  Error Free
5     Error Free  Error Free  Error Free  Error Free
6     Error Free   Defective  Error Free  Error Free
7     Error Free  Error Free  Error Free  Error Free
8     Error Free  Error Free  Error Free  Error Free
9     Error Free  Error Free  Error Free  Error Free
10     Defective  Error Free  Error Free  Error Free
11    Error Free  Error Free  Error Free  Error Free
12    Error Free  Error Free  Error Free   Defective
13    Error Free  Error Free  Error Free  Error Free
14    Error Free  Error Free  Error Free  Error Free
15    Error Free  Error Free   Defective  Error Free
16    Error Free  Error Free  Error Free  Error Free
17    Error Free  Error Free  Error Free  Error Free
18    Error Free  Error Free  Error Free  Error Free
19     Defective  Error Free  Error Free   Defective
20    Error Free  Error Free  Error Free  Error Free
21    Error Free  Error Free   Defective  Error Free
22    Error Free  Error Free  Error Free  Error Free
23     Defective  Error Free  Error Free  Error Free
24    Error Free  Error Free  Error Free  Error Free
25    Error Free  Error Free  Error Free  Error Free
26    Error Free  Error Free  Error Free  Error Free
27    Error Free  Error Free  Error Free  Error Free
28    Error Free  Error Free  Error Free  Error Free
29    Error Free  Error Free  Error Free  Error Free
30    Error Free  Error Free  Error Free  Error Free
31    Error Free  Error Free  Error Free   Defective
32    Error Free  Error Free  Error Free  Error Free
33     Defective  Error Free  Error Free  Error Free
34    Error Free  Error Free  Error Free  Error Free
35    Error Free  Error Free  Error Free  Error Free
36    Error Free  Error Free  Error Free  Error Free
37    Error Free  Error Free   Defective  Error Free
38    Error Free  Error Free  Error Free  Error Free
39     Defective  Error Free   Defective  Error Free
40    Error Free  Error Free  Error Free  Error Free
41     Defective  Error Free  Error Free  Error Free
42    Error Free  Error Free  Error Free  Error Free
43    Error Free  Error Free  Error Free  Error Free
44     Defective  Error Free   Defective  Error Free
45    Error Free  Error Free  Error Free  Error Free
46    Error Free  Error Free  Error Free  Error Free
47     Defective  Error Free  Error Free  Error Free
48    Error Free  Error Free  Error Free  Error Free
49    Error Free  Error Free  Error Free   Defective
50    Error Free  Error Free  Error Free  Error Free
51    Error Free   Defective  Error Free  Error Free
52    Error Free  Error Free  Error Free  Error Free
53    Error Free  Error Free  Error Free  Error Free
54    Error Free  Error Free  Error Free  Error Free
55    Error Free  Error Free  Error Free  Error Free
56    Error Free  Error Free  Error Free  Error Free
57    Error Free  Error Free  Error Free  Error Free
58    Error Free  Error Free  Error Free  Error Free
59     Defective  Error Free  Error Free  Error Free
60    Error Free  Error Free  Error Free   Defective
61    Error Free  Error Free  Error Free  Error Free
62    Error Free  Error Free  Error Free  Error Free
63    Error Free  Error Free  Error Free  Error Free
64    Error Free  Error Free  Error Free  Error Free
65    Error Free  Error Free  Error Free  Error Free
66    Error Free  Error Free   Defective  Error Free
67    Error Free  Error Free  Error Free  Error Free
68    Error Free   Defective  Error Free  Error Free
69    Error Free  Error Free  Error Free  Error Free
70    Error Free  Error Free  Error Free  Error Free
71     Defective   Defective  Error Free  Error Free
72    Error Free  Error Free  Error Free  Error Free
73    Error Free  Error Free   Defective  Error Free
74    Error Free  Error Free 

H0 = There is no defects in Order Forms

H1 = There is a defects in Order Forms

In [42]:
customer.Phillippines.value_counts(), '\n' ,customer.Indonesia.value_counts(), '\n' ,customer.Malta.value_counts(), '\n' ,customer.India.value_counts()

(Error Free    271
 Defective      29
 Name: Phillippines, dtype: int64,
 '\n',
 Error Free    267
 Defective      33
 Name: Indonesia, dtype: int64,
 '\n',
 Error Free    269
 Defective      31
 Name: Malta, dtype: int64,
 '\n',
 Error Free    280
 Defective      20
 Name: India, dtype: int64)

In [44]:
import numpy as np
data = np.array([[271,267,269,280],[29,33,31,20]])
data

array([[271, 267, 269, 280],
       [ 29,  33,  31,  20]])

In [45]:
chi2_contingency(data)

(3.858960685820355,
 0.2771020991233135,
 3,
 array([[271.75, 271.75, 271.75, 271.75],
        [ 28.25,  28.25,  28.25,  28.25]]))

Chi-square value = 3.8589

p_value = 0.2771

degree of freedom = 3

alpha value = 0.05

As the p_value is less than alpha value we rejecting the null hypothesis.

So we can tell there is defects in the Customer Order Forms.